lib/WUI/wui_api.h

```diff
@@ -186,14 +186,14 @@ uint8_t dhcp_addrs_are_supplied(void);
 /*!****************************************************************************
 * \brief Parses time from device's time storage in dest string in format hh:mm:ss
 *
-* \param dest - destination string for paresd time ! At least 10 chars !
+* \param dest - destination string for paresd time ! At least 12 chars !
 *****************************************************************************/
 void sntp_get_system_time(char * dest);
 
 /*!****************************************************************************
 * \brief Parses date from device's time storage in dest string in format dd.mm.yyyy
 *
-* \param dest - destination string for paresd date ! At least 12 chars !
+* \param dest - destination string for paresd date ! At least 13 chars !
 *****************************************************************************/
 void sntp_get_system_date(char * dest);
```

Comment:
It is not a good idea to rely on comment checking out of array bounds access.

You can enforce exact size array in function prototype.
e.g.
in header:
typedef char system_time_t[12];
void sntp_get_system_time(system_time_t *system_time);

You need to change also implementation accordingly, as now parameter is not type of pointer to char but pointer to char[12] (pointer to pointer to char).

void sntp_get_system_time(system_time_t *system_time) {

    if (sntp_time_init) {
        RTC_TimeTypeDef currTime;

        HAL_RTC_GetTime(&hrtc, &currTime, RTC_FORMAT_BIN);

        snprintf(*system_time, sizeof(system_time_t), "%02d:%02d:%02d", currTime.Hours, currTime.Minutes, currTime.Seconds);
    } else {
        strcpy(*system_time, "N/A");
    }
}

note: You can NOT avoid pointer to pointer as if you would change function prototype only to:
void sntp_get_system_time(system_time_t system_time);
this would decay to char * and your compile time check is lost. (It is by definition of C programming language, that arrays decays to pointers and array size information is lost in this process.)

Reply:
good point!

Reply:
commit `a26ae3f`